core/docker-entrypoint.sh

```diff
@@ -38,16 +38,23 @@ if [ "$IGNORE_DATA_ROOT_MOUNT_CHECK" = "false" ] && mount | grep "/root/.questdb
     QUESTDB_DATA_DIR="/root/.questdb"
 fi
 
+# Check if on-demand JVM arguments are provided through environment variable
+if [ -n "$JVM_PREPEND" ]; then
+    echo "Found on-demand JVM arguments: $JVM_PREPEND, prepending to JVM args"
+    JAVA_COMMAND="$JAVA_COMMAND $JVM_PREPEND"
+fi
+
+# Check if arguments are provided in the configuration file
 if [ $# -eq 0 ]; then
-    echo "No arguments found, start with default arguments"
-    set -- /app/bin/java -XX:ErrorFile=${QUESTDB_DATA_DIR}/db/hs_err_pid+%p.log -Dout=${QUESTDB_DATA_DIR}/conf/log.conf -m io.questdb/io.questdb.ServerMain -d ${QUESTDB_DATA_DIR} -f
+    echo "No arguments found in the configuration, start with default arguments"
+    set -- $JAVA_COMMAND -XX:ErrorFile=${QUESTDB_DATA_DIR}/db/hs_err_pid+%p.log -Dout=${QUESTDB_DATA_DIR}/conf/log.conf -m io.questdb/io.questdb.ServerMain -d ${QUESTDB_DATA_DIR} -f
 else
     if [ "${1:0:1}" = '-' ]; then
         echo "Found config arguments $@"
-        set -- /app/bin/java "$@"
+        set -- $JAVA_COMMAND "$@"
     elif [ "$1" = "/app/bin/java" ]; then
         echo "Java binary arguments found, Non default arguments config run"
-        set -- "$@"
+        set -- $JVM_PREPEND "$@"
```

Comment:
Let's keep the old command here. The use case is `docker run <image-id> /app/bin/java the_remaining_args` - so, full control over the startup command.

If we use `set -- $JVM_PREPEND "$@"` here, command like `docker run -e JVM_PREPEND="-Xmx5g" 5caa247a98b1 /app/bin/java` leads to `gosu 10001:10001 -Xmx5g /app/bin/java` which has incorrect flag order.

Reply:
Thanks for clarifying this use-case! Will fix